### DISCUSSION TITLE
How to transitively copy files to output across NuGet packages?

### DISCUSSION THREAD

**Discussion by anslmorvan** [NuGet/Home#12523] [Q&A] [answered]
Hello, I'm trying something that looks simply but I didn't succeed in making it work.

I have a `PackageA` that needs a file to be copied to the output of the final application to work as expected.
I include the needed file into my `PackageA`'s project with these lines:
```xml
  <ItemGroup>
    <Content Include="test.txt">
      <PackageCopyToOutput>true</PackageCopyToOutput>
    </Content>
  </ItemGroup>
```

If I consume the generated package directly, the file `test.txt` is copied to the output.

Now, if I reference this `PackageA` from another `PackageB`, and reference only the `PackageB` rather than `PackageA`, the file `test.txt` is not copied at all to the output while I expected this file to be transitively copied.

I tried digging into the issues of this repository and found that we should play with the `PrivateAssets` or something like that (#12270 and all its siblings). But I never succeeded in making this work.

I made a [github repository to show off the issue I'm facing](https://github.com/lmorvan/nuget-transitively-copy-to-output).

I also know that it is possible to copy the file by including `.props` or a `.targets` file doing so with a pre-build or post-build action, but I find this solution quite hacky, not elegant, and hard to set up and maintain over time. Moreover, I do have the feeling that NuGet is supposed to support this kind of usage quite easily, so maybe I have missed something somewhere.

Can you help me? 😄 

PS: I just added project publishing in my sample repository. It doesn't work better.
PS: Question also asked on StackOverflow: https://stackoverflow.com/questions/75929069/how-to-transitively-copy-files-to-output-across-nuget-packages

**Comment by msedi** [comment]
I would also be interested in this topic. It seems that many things have been made for transitivity (e.g. buildTransitive), but it seems content files have been missed. The solution that has been found in #6151 works but it is something consumers have to know and to remember. 

I see a nuget package as something like a `ProjectReference`. With a  `ProjectReference` setting the `Content` to       `<CopyToOutputDirectory>Always</CopyToOutputDirectory>` works always transitively. So it would be nice if something similar could be made possible here.

**Comment by anslmorvan** [comment, marked as answer]
We have found a solution by placing the files to copy into the `runtimes/any/native` folder. But it requires you to `publish` the final application or at least set a `RuntimeIdentifier` to the executable project if you just want to `build`.
